configmanager/env.go

```diff
@@ -0,0 +1,256 @@
+	var zero T
+
+	result, err := parseBool(valueStr)
+	if err != nil {
+		return zero, err
+	}
+
+	return any(result).(T), nil
+}
+
+// parseBool parses boolean values (accepts "true" or "false")
+func parseBool(valueStr string) (bool, error) {
+	valueStr = strings.ToLower(strings.TrimSpace(valueStr))
+
+	switch valueStr {
+	case "true":
+		return true, nil
+	case "false":
+		return false, nil
+	default:
+		return false, fmt.Errorf("invalid boolean value: %s (must be 'true' or 'false')", valueStr)
+	}
+}
+
+// EnvVarSpec defines a specification for reading an environment variable.
+// All fields except Name are optional.
+//
+// Example usage:
+//
+//	specs := []configmanager.EnvVarSpec{
+//	    {Name: "DATABASE_URL"},  // Required by default
+//	    {Name: "PORT", Optional: true, DefaultValue: "5432"},  // Optional with default
+//	    {Name: "API_KEY", Description: "API authentication key"},  // Required with description
+//	}
+//	envVars, errors := configmanager.ReadEnvVars(specs)
+//	if len(errors) > 0 {
+//	    return fmt.Errorf("missing required vars: %v", errors)
+//	}
+type EnvVarSpec struct {
+	Name         string // Required: The environment variable name to read
+	Optional     bool   // Optional: If true, env var is optional; if false, it's required (default: false/required)
+	DefaultValue string // Optional: Value to use when env var is not set (default: "")
+	Description  string // Optional: Human-readable description for error messages (default: Name)
+}
+
+// ReadEnvVars reads multiple environment variables based on the provided specifications.
+// Returns a map of environment variable names to their values and a slice of errors.
+// Environment variables are required by default unless Optional is set to true.
+//
+// Example usage:
+//
+//	specs := []configmanager.EnvVarSpec{
+//	    {Name: "MONGODB_URI", Description: "MongoDB connection URI"},
+//	    {Name: "MONGODB_DATABASE_NAME"},
+//	    {Name: "MONGODB_PORT", Optional: true, DefaultValue: "27017"},
+//	}
+//	envVars, errors := configmanager.ReadEnvVars(specs)
+//	if len(errors) > 0 {
+//	    log.Fatalf("Missing required environment variables: %v", errors)
+//	}
+//	// Use the values
+//	dbURI := envVars["MONGODB_URI"]
+//	dbName := envVars["MONGODB_DATABASE_NAME"]
+//	dbPort := envVars["MONGODB_PORT"]  // Will be "27017" if not set
+func ReadEnvVars(specs []EnvVarSpec) (map[string]string, []error) {
+	results := make(map[string]string)
+
+	var errors []error
+
+	for _, spec := range specs {
+		value, exists := os.LookupEnv(spec.Name)
+
+		if !exists {
+			value, err := handleMissingEnvVar(spec)
+			if err != nil {
+				errors = append(errors, err)
+			}
+
+			if value != "" || spec.Optional {
+				results[spec.Name] = value
+			}
+
+			continue
+		}
+
+		results[spec.Name] = value
+	}
+
+	return results, errors
+}
+
+func handleMissingEnvVar(spec EnvVarSpec) (string, error) {
+	if spec.Optional {
+		return spec.DefaultValue, nil
+	}
+
+	description := spec.Description
```

Comment:
What is the use case for description? Can't these be simple code comments? Do we have logic that hangs off of the description? If we do, then it will be very brittle